File: funcscript/funcs/logic/AndFunction.cs
using funcscript.core;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using funcscript.model;

namespace funcscript.funcs.logic
{

    public class AndFunction : IFsFunction
    {
        public int MaxParsCount => -1;

        public CallType CallType => CallType.Infix;

        public string Symbol => "and";

        public int Precidence => 400;

        public object Evaluate(IFsDataProvider parent, IParameterList pars)
        {
            int count = pars.Count;

            for (int i = 0; i < count; i++)
            {
                var par = pars.GetParameter(parent, i);

                if (!(par is bool b))
                    return new FsError(FsError.ERROR_TYPE_MISMATCH,
                        $"{this.Symbol} doesn't apply to this type:{(par == null ? "null" : par.GetType())} ");

                if(!b)
                    return false;
            }

            return true;
        }


        public string ParName(int index)
        {
            return $"Value {index + 1}";
        }
    }
}

File: js_port/fs_js/src/funcs/logic/AndFunction.js
const { BaseFunction, CallType } = require('../../core/functionBase');
const helpers = require('../helpers');

class AndFunction extends BaseFunction {
  constructor() {
    super();
    this.symbol = 'and';
    this.callType = CallType.Infix;
    this.precidence = 400;
  }

  get maxParameters() {
    return -1;
  }

  evaluate(provider, parameters) {
    const count = parameters.count;
    for (let i = 0; i < count; i += 1) {
      const param = parameters.getParameter(provider, i);
      if (helpers.typeOf(param) !== helpers.FSDataType.Boolean) {
        return helpers.makeError(
          helpers.FsError.ERROR_TYPE_MISMATCH,
          `${this.symbol} doesn't apply to this type: ${helpers.typeOf(param)}`
        );
      }
      if (!helpers.valueOf(param)) {
        return helpers.makeValue(helpers.FSDataType.Boolean, false);
      }
    }
    return helpers.makeValue(helpers.FSDataType.Boolean, true);
  }
}

module.exports = {
  AndFunction
};
